libs/application/templates/aosh/street-registration/src/fields/MachinesField/MachineSelectField.tsx

```diff
@@ -0,0 +1,193 @@
+import { FieldBaseProps, Option } from '@island.is/application/types'
+import { useLocale } from '@island.is/localization'
+import { FC, useCallback, useEffect, useState } from 'react'
+import {
+  AlertMessage,
+  Box,
+  Bullet,
+  BulletList,
+  SkeletonLoader,
+  InputError,
+  ActionCard,
+} from '@island.is/island-ui/core'
+import { information, error } from '../../lib/messages'
+import { SelectController } from '@island.is/shared/form-fields'
+import { useLazyMachineDetails } from '../../hooks/useLazyMachineDetails'
+import { useFormContext } from 'react-hook-form'
+import { getValueViaPath } from '@island.is/application/core'
+import { MachineDto } from '@island.is/clients/work-machines'
+import { mustInspectBeforeStreetRegistration } from '../../utils/getSelectedMachine'
+
+interface MachineSearchFieldProps {
+  currentMachineList: MachineDto[]
+}
+
+export const MachineSelectField: FC<
```

Comment:
Yes, I agree with you.

But I don't think it should be part of this PR as it would require extensive testing for all the different applications.
We will add to our todos